### DISCUSSION TITLE
Re-sending Grasp Poses and Manual Operation Queries

### DISCUSSION THREAD

**Discussion by AbbeGo** [Victorlouisdg/cloth-competition#28] [Questions] [answered]
After the planning process fails for the initial grasp pose sent, how do we resend a new grasp pose? Additionally, is it possible to manually operate the download of observations? If so, can the sending of poses also be done manually? 
If you can answer this question promptly, I will be very grateful!

**Comment by Victorlouisdg** [comment, marked as answer]
Thanks for your questions! Here's the breakdown:

**Question 1: How can we resend a new grasp pose?**
You can send multiple grasp poses by calling the `upload_grasp()` function repeatedly. I'll store all submitted grasps and try planning them from oldest to newest. Prioritize sending your most confident grasp first, followed by safer alternatives.

**Question 2: Can we manually download or upload?**
Manual uploads and downloads aren't supported, and I won’t be able to implement this in time. Please stick to the functions provided in the `04_server_communication.ipynb` notebook. This helps avoid potential errors like using an incorrect team name or sample ID.

**Reply by AbbeGo** [reply]
Thank you for your prompt response! Apologies for any confusion. Regarding question 2, it seems there was a discrepancy between their response and what I intended to ask. I was wondering if it's acceptable for the team to restart the code to download observations before testing each sample, or if we should refrain from restarting between testing samples and keep the code in a waiting state.

**Reply by Victorlouisdg** [reply]
No problem, I indeed misunderstood the question. Yes restarting your code and downloading the observation again is acceptable, but you cannot improve or assist in the choice of grasp in any way.  

See this post for more details: https://github.com/Victorlouisdg/cloth-competition/discussions/24

**Reply by AbbeGo** [reply]
Thank you for your reply! If we use a notebook to complete the competition, is preloading the model before downloading the observation folder allowed?

**Reply by Victorlouisdg** [reply]
Yes, that is allowed.
